homeassistant/components/magichome/__init__.py

```diff
@@ -0,0 +1,160 @@
+"""Support for MagicHome Smart devices."""
+from datetime import timedelta
+import logging
+import voluptuous as vol
+
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD, CONF_PLATFORM
+from homeassistant.helpers import discovery
+from homeassistant.helpers.dispatcher import dispatcher_send, async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import track_time_interval
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_COMPANY = "company"
+
+DOMAIN = "magichome"
+DATA_MAGICHOME = "data_magichome"
+
+SIGNAL_DELETE_ENTITY = "magichome_delete"
+SIGNAL_UPDATE_ENTITY = "magichome_update"
+
+SERVICE_FORCE_UPDATE = "force_update"
+SERVICE_PULL_DEVICES = "pull_devices"
+
+MAGICHOME_TYPE_TO_HA = {
+    "light": "light",
+    "scene": "scene",
+    "switch": "switch",
+    "socket": "socket",
+    "controller": "controller",
+}
```

Comment:
This looks like a 1-to-1 mapping.

Reply:
Yes, does this need to be modified?

Reply:
Yes this needs to be modified since 1-to-1 mapping is redundant.

Reply:
@LIULi-VVET this still needs to be changed

Reply:
I would recommend to use a list instead of a dictionary and remove line 70.